src/mixer/audio.ts

```
@@ -156,15 +166,30 @@ class Player extends ((PlayerEmitter as unknown) as { new (): EventEmitter }) {
   _applyVolume() {
     const level = this.volume + this.trim;
     const linear = Math.pow(10, level / 20);
-    if (linear < 1) {
-      this.wavesurfer.setVolume(linear);
-      if (!this.customOutput) {
-        (this.wavesurfer as any).backend.gainNode.gain.value = 1;
-      }
+
+    // Actually adjust the wavesurfer gain node gain instead, so we can tap off analyser for PFL.
+    this.wavesurfer.setVolume(1);
+    if (!this.customOutput) {
+      (this.wavesurfer as any).backend.gainNode.gain.value = linear;
+    }
+  }
+
+  _connectPFL() {
+    if (this.pfl) {
+      console.log("Connecting PFL");
+
+      // In this case, we just want to route the player output to the headphones direct.
+      // Tap it from analyser to avoid the player volume.
+      (this.wavesurfer as any).backend.analyser.connect(
+        this.engine.headphonesNode
+      );
     } else {
-      this.wavesurfer.setVolume(1);
-      if (!this.customOutput) {
-        (this.wavesurfer as any).backend.gainNode.gain.value = linear;
+      try {
+        (this.wavesurfer as any).backend.analyser.disconnect(
+          this.engine.headphonesNode
+        );
+      } catch (e) {
+        console.log("Didn't disconnect.");
```

Not a fan of swallowing exceptions, this should at least get sent to Raygun (which a `console.warn(e)` would do).

---

Relabled to explain why.